From efa2361d1e305ee50cc2bf3180945ecfe829c0de Mon Sep 17 00:00:00 2001
From: Joe Zhang <joe.zhang@technexion.com>
Date: Fri, 11 Oct 2024 17:51:49 +0800
Subject: [PATCH 5/5] fix issue failed to boot kernel itb image

CONFIG_SYS_BOOTM_LEN: from 0x2000000 to 0x4000000

Additional configs for imx9 series:
CONFIG_FIT_EXTERNAL_OFFSET=0x3000
CONFIG_SPL_LOAD_FIT=y
---
 configs/axon-imx8mp_defconfig                                  | 2 +-
 configs/axon-imx91_defconfig                                   | 2 ++
 configs/axon-imx93_defconfig                                   | 2 ++
 configs/edm-g-imx8mp_defconfig                                 | 2 +-
 configs/edm-imx93_defconfig                                    | 2 ++
 configs/imx8mm-cl-iot-gate-optee_defconfig                     | 2 +-
 configs/imx8mm-cl-iot-gate_defconfig                           | 2 +-
 configs/imx8mn_ab2_defconfig                                   | 2 +-
 configs/imx8mn_bsh_smm_s2_defconfig                            | 2 +-
 configs/imx8mn_bsh_smm_s2pro_defconfig                         | 2 +-
 configs/imx8mn_ddr3l_ab2_defconfig                             | 2 +-
 configs/imx8mn_ddr3l_evk_defconfig                             | 2 +-
 configs/imx8mn_ddr4_ab2_defconfig                              | 2 +-
 configs/imx8mn_ddr4_evk_android_defconfig                      | 2 +-
 configs/imx8mn_ddr4_evk_android_uuu_defconfig                  | 2 +-
 configs/imx8mn_ddr4_evk_defconfig                              | 2 +-
 configs/imx8mn_ddr4_evk_ld_defconfig                           | 2 +-
 configs/imx8mn_evk_android_defconfig                           | 2 +-
 configs/imx8mn_evk_android_dual_defconfig                      | 2 +-
 configs/imx8mn_evk_android_trusty_defconfig                    | 2 +-
 configs/imx8mn_evk_android_trusty_dual_defconfig               | 2 +-
 configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig | 2 +-
 configs/imx8mn_evk_android_uuu_defconfig                       | 2 +-
 configs/imx8mn_evk_defconfig                                   | 2 +-
 configs/imx8mn_evk_ld_defconfig                                | 2 +-
 configs/imx8mn_var_som_defconfig                               | 2 +-
 configs/imx8mp-icore-mx8mp-edimm2.2_defconfig                  | 2 +-
 configs/imx8mp_beacon_defconfig                                | 2 +-
 configs/imx8mp_ddr4_evk_defconfig                              | 2 +-
 configs/imx8mp_ddr4_evk_inline_ecc_defconfig                   | 2 +-
 configs/imx8mp_ddr4_evk_nand_defconfig                         | 2 +-
 configs/imx8mp_debix_model_a_defconfig                         | 2 +-
 configs/imx8mp_evk_android_defconfig                           | 2 +-
 configs/imx8mp_evk_android_dual_defconfig                      | 2 +-
 configs/imx8mp_evk_android_powersave_defconfig                 | 2 +-
 configs/imx8mp_evk_android_trusty_defconfig                    | 2 +-
 configs/imx8mp_evk_android_trusty_dual_defconfig               | 2 +-
 configs/imx8mp_evk_android_trusty_powersave_dual_defconfig     | 2 +-
 configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig | 2 +-
 configs/imx8mp_evk_android_uuu_defconfig                       | 2 +-
 configs/imx8mp_evk_defconfig                                   | 2 +-
 configs/imx8mp_evk_inline_ecc_defconfig                        | 2 +-
 configs/imx8mp_evk_ndm_defconfig                               | 2 +-
 configs/imx8mp_rsb3720a1_4G_defconfig                          | 2 +-
 configs/imx8mp_rsb3720a1_6G_defconfig                          | 2 +-
 configs/imx8mq_cm_defconfig                                    | 2 +-
 configs/imx91_11x11_evk_defconfig                              | 2 ++
 configs/imx91_11x11_evk_inline_ecc_defconfig                   | 2 ++
 configs/imx91_11x11_evk_spinand_defconfig                      | 2 ++
 configs/imx93_11x11_evk_android_defconfig                      | 2 ++
 configs/imx93_11x11_evk_android_dual_defconfig                 | 2 ++
 configs/imx93_11x11_evk_android_trusty_defconfig               | 2 ++
 configs/imx93_11x11_evk_android_trusty_dual_defconfig          | 2 ++
 configs/imx93_11x11_evk_android_uuu_defconfig                  | 2 ++
 configs/imx93_11x11_evk_defconfig                              | 2 ++
 configs/imx93_11x11_evk_matter_trusty_defconfig                | 2 ++
 configs/imx93_11x11_evk_pf0900_defconfig                       | 2 ++
 configs/kontron_pitx_imx8m_defconfig                           | 2 +-
 configs/msc_sm2s_imx8mp_defconfig                              | 2 +-
 configs/pico-imx8mq_defconfig                                  | 2 +-
 configs/pico-imx93_defconfig                                   | 2 ++
 configs/tek-imx8mp_defconfig                                   | 2 +-
 configs/tep-imx8mp_defconfig                                   | 2 +-
 63 files changed, 78 insertions(+), 48 deletions(-)

diff --git a/configs/axon-imx8mp_defconfig b/configs/axon-imx8mp_defconfig
index c3755b21975..4225fdaec29 100644
--- a/configs/axon-imx8mp_defconfig
+++ b/configs/axon-imx8mp_defconfig
@@ -29,7 +29,7 @@ CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/axon-imx91_defconfig b/configs/axon-imx91_defconfig
index 23dea6b1d65..9dd25e52040 100644
--- a/configs/axon-imx91_defconfig
+++ b/configs/axon-imx91_defconfig
@@ -30,6 +30,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_DISTRO_DEFAULTS=y
diff --git a/configs/axon-imx93_defconfig b/configs/axon-imx93_defconfig
index 14625368af9..df9b45a7bbd 100644
--- a/configs/axon-imx93_defconfig
+++ b/configs/axon-imx93_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_DISTRO_DEFAULTS=y
diff --git a/configs/edm-g-imx8mp_defconfig b/configs/edm-g-imx8mp_defconfig
index 125c96f422a..21449a8e90a 100644
--- a/configs/edm-g-imx8mp_defconfig
+++ b/configs/edm-g-imx8mp_defconfig
@@ -29,7 +29,7 @@ CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/edm-imx93_defconfig b/configs/edm-imx93_defconfig
index b2095eaadf1..5d972a1217e 100644
--- a/configs/edm-imx93_defconfig
+++ b/configs/edm-imx93_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_DISTRO_DEFAULTS=y
diff --git a/configs/imx8mm-cl-iot-gate-optee_defconfig b/configs/imx8mm-cl-iot-gate-optee_defconfig
index c011f4f4906..2e2bcbd1913 100644
--- a/configs/imx8mm-cl-iot-gate-optee_defconfig
+++ b/configs/imx8mm-cl-iot-gate-optee_defconfig
@@ -22,7 +22,7 @@ CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_FIT_SIGNATURE=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_CBSIZE=2048
diff --git a/configs/imx8mm-cl-iot-gate_defconfig b/configs/imx8mm-cl-iot-gate_defconfig
index 6afdb399b14..71a73d93969 100644
--- a/configs/imx8mm-cl-iot-gate_defconfig
+++ b/configs/imx8mm-cl-iot-gate_defconfig
@@ -24,7 +24,7 @@ CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_FIT_SIGNATURE=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_CBSIZE=2048
diff --git a/configs/imx8mn_ab2_defconfig b/configs/imx8mn_ab2_defconfig
index 77947f983e3..41446213eba 100644
--- a/configs/imx8mn_ab2_defconfig
+++ b/configs/imx8mn_ab2_defconfig
@@ -55,7 +55,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_bsh_smm_s2_defconfig b/configs/imx8mn_bsh_smm_s2_defconfig
index 8c28b92f752..66a821f0d5f 100644
--- a/configs/imx8mn_bsh_smm_s2_defconfig
+++ b/configs/imx8mn_bsh_smm_s2_defconfig
@@ -23,7 +23,7 @@ CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DEFAULT_FDT_FILE="freescale/imx8mn-bsh-smm-s2.dtb"
diff --git a/configs/imx8mn_bsh_smm_s2pro_defconfig b/configs/imx8mn_bsh_smm_s2pro_defconfig
index 3b55f6ff418..ba427a70143 100644
--- a/configs/imx8mn_bsh_smm_s2pro_defconfig
+++ b/configs/imx8mn_bsh_smm_s2pro_defconfig
@@ -24,7 +24,7 @@ CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DEFAULT_FDT_FILE="freescale/imx8mn-bsh-smm-s2pro.dtb"
diff --git a/configs/imx8mn_ddr3l_ab2_defconfig b/configs/imx8mn_ddr3l_ab2_defconfig
index 5b9487a1e11..8aad4023598 100644
--- a/configs/imx8mn_ddr3l_ab2_defconfig
+++ b/configs/imx8mn_ddr3l_ab2_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr3l_evk_defconfig b/configs/imx8mn_ddr3l_evk_defconfig
index 9130f53fb74..3c1d7a68703 100644
--- a/configs/imx8mn_ddr3l_evk_defconfig
+++ b/configs/imx8mn_ddr3l_evk_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_ddr4_ab2_defconfig b/configs/imx8mn_ddr4_ab2_defconfig
index 5746d82a293..92979162b93 100644
--- a/configs/imx8mn_ddr4_ab2_defconfig
+++ b/configs/imx8mn_ddr4_ab2_defconfig
@@ -51,7 +51,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="u-boot=> "
diff --git a/configs/imx8mn_ddr4_evk_android_defconfig b/configs/imx8mn_ddr4_evk_android_defconfig
index bc1011495f6..d0d1d4cc6e8 100644
--- a/configs/imx8mn_ddr4_evk_android_defconfig
+++ b/configs/imx8mn_ddr4_evk_android_defconfig
@@ -50,7 +50,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr4_evk_android_uuu_defconfig b/configs/imx8mn_ddr4_evk_android_uuu_defconfig
index 5d7afd6b635..1a43f97b79f 100644
--- a/configs/imx8mn_ddr4_evk_android_uuu_defconfig
+++ b/configs/imx8mn_ddr4_evk_android_uuu_defconfig
@@ -51,7 +51,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr4_evk_defconfig b/configs/imx8mn_ddr4_evk_defconfig
index 5ad4a29b6af..8c642aae167 100644
--- a/configs/imx8mn_ddr4_evk_defconfig
+++ b/configs/imx8mn_ddr4_evk_defconfig
@@ -51,7 +51,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_ddr4_evk_ld_defconfig b/configs/imx8mn_ddr4_evk_ld_defconfig
index 1d4ccacd89b..ac1d1af1588 100644
--- a/configs/imx8mn_ddr4_evk_ld_defconfig
+++ b/configs/imx8mn_ddr4_evk_ld_defconfig
@@ -57,7 +57,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mn_evk_android_defconfig b/configs/imx8mn_evk_android_defconfig
index fc41b3f7624..1f5ac90ff37 100644
--- a/configs/imx8mn_evk_android_defconfig
+++ b/configs/imx8mn_evk_android_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_android_dual_defconfig b/configs/imx8mn_evk_android_dual_defconfig
index 9f9bf0a0a99..ce731c98434 100644
--- a/configs/imx8mn_evk_android_dual_defconfig
+++ b/configs/imx8mn_evk_android_dual_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_defconfig b/configs/imx8mn_evk_android_trusty_defconfig
index b65cc932518..134e696c4d2 100644
--- a/configs/imx8mn_evk_android_trusty_defconfig
+++ b/configs/imx8mn_evk_android_trusty_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_dual_defconfig b/configs/imx8mn_evk_android_trusty_dual_defconfig
index 3ef083092ff..d9a10acc127 100644
--- a/configs/imx8mn_evk_android_trusty_dual_defconfig
+++ b/configs/imx8mn_evk_android_trusty_dual_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig b/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
index 1008cbd420d..dab731735b9 100644
--- a/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
+++ b/configs/imx8mn_evk_android_trusty_secure_unlock_dual_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_android_uuu_defconfig b/configs/imx8mn_evk_android_uuu_defconfig
index 9dcbfa2fce1..2ef78161431 100644
--- a/configs/imx8mn_evk_android_uuu_defconfig
+++ b/configs/imx8mn_evk_android_uuu_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_defconfig b/configs/imx8mn_evk_defconfig
index 6bd43aa85d8..99bc202cb5a 100644
--- a/configs/imx8mn_evk_defconfig
+++ b/configs/imx8mn_evk_defconfig
@@ -55,7 +55,7 @@ CONFIG_SPL_I2C=y
 CONFIG_SPL_POWER=y
 CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_evk_ld_defconfig b/configs/imx8mn_evk_ld_defconfig
index 498a62de5e7..8c17b0e3b84 100644
--- a/configs/imx8mn_evk_ld_defconfig
+++ b/configs/imx8mn_evk_ld_defconfig
@@ -61,7 +61,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mn_var_som_defconfig b/configs/imx8mn_var_som_defconfig
index 6057d13a73d..a51071af42f 100644
--- a/configs/imx8mn_var_som_defconfig
+++ b/configs/imx8mn_var_som_defconfig
@@ -26,7 +26,7 @@ CONFIG_OF_BOARD_FIXUP=y
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig b/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
index 83d7d18f316..d92adb0ad17 100644
--- a/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
+++ b/configs/imx8mp-icore-mx8mp-edimm2.2_defconfig
@@ -25,7 +25,7 @@ CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DEFAULT_FDT_FILE="imx8mp-icore-mx8mp-edimm2.2.dtb"
diff --git a/configs/imx8mp_beacon_defconfig b/configs/imx8mp_beacon_defconfig
index 9d09dd4a7ce..f93c906c121 100644
--- a/configs/imx8mp_beacon_defconfig
+++ b/configs/imx8mp_beacon_defconfig
@@ -31,7 +31,7 @@ CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_BOOTCOMMAND="mmc dev ${mmcdev}; if mmc rescan; then if run loadbootscript; then run bootscript; else if run loadimage; then run mmcboot; else run netboot; fi; fi; else booti ${loadaddr} - ${fdt_addr}; fi"
diff --git a/configs/imx8mp_ddr4_evk_defconfig b/configs/imx8mp_ddr4_evk_defconfig
index e2113e8876f..192e9296d59 100644
--- a/configs/imx8mp_ddr4_evk_defconfig
+++ b/configs/imx8mp_ddr4_evk_defconfig
@@ -53,7 +53,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_ddr4_evk_inline_ecc_defconfig b/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
index 96a17e1f26b..26aed43683b 100644
--- a/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
+++ b/configs/imx8mp_ddr4_evk_inline_ecc_defconfig
@@ -53,7 +53,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_ddr4_evk_nand_defconfig b/configs/imx8mp_ddr4_evk_nand_defconfig
index 7446f7d613a..73ddeef4066 100644
--- a/configs/imx8mp_ddr4_evk_nand_defconfig
+++ b/configs/imx8mp_ddr4_evk_nand_defconfig
@@ -51,7 +51,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_debix_model_a_defconfig b/configs/imx8mp_debix_model_a_defconfig
index 5092ac22744..5c78e35c59b 100644
--- a/configs/imx8mp_debix_model_a_defconfig
+++ b/configs/imx8mp_debix_model_a_defconfig
@@ -22,7 +22,7 @@ CONFIG_SYS_LOAD_ADDR=0x40480000
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DEFAULT_FDT_FILE="imx8mp-debix-model-a.dtb"
diff --git a/configs/imx8mp_evk_android_defconfig b/configs/imx8mp_evk_android_defconfig
index 751a5fc0284..ff5eb2e7f7f 100644
--- a/configs/imx8mp_evk_android_defconfig
+++ b/configs/imx8mp_evk_android_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_dual_defconfig b/configs/imx8mp_evk_android_dual_defconfig
index 33997b2a021..30d10cc6ea1 100644
--- a/configs/imx8mp_evk_android_dual_defconfig
+++ b/configs/imx8mp_evk_android_dual_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_powersave_defconfig b/configs/imx8mp_evk_android_powersave_defconfig
index 8466d1057a4..6690055f67d 100644
--- a/configs/imx8mp_evk_android_powersave_defconfig
+++ b/configs/imx8mp_evk_android_powersave_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_defconfig b/configs/imx8mp_evk_android_trusty_defconfig
index e6d168c4364..33d2359c181 100644
--- a/configs/imx8mp_evk_android_trusty_defconfig
+++ b/configs/imx8mp_evk_android_trusty_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_dual_defconfig b/configs/imx8mp_evk_android_trusty_dual_defconfig
index e315abc5ef7..a9b675b1f40 100644
--- a/configs/imx8mp_evk_android_trusty_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_dual_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig b/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
index 1d5cc2bb153..56a0fb8251f 100644
--- a/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_powersave_dual_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig b/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
index a3202897de1..7bb28bd2138 100644
--- a/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
+++ b/configs/imx8mp_evk_android_trusty_secure_unlock_dual_defconfig
@@ -57,7 +57,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_android_uuu_defconfig b/configs/imx8mp_evk_android_uuu_defconfig
index 12a5774754d..9a14b9b74d5 100644
--- a/configs/imx8mp_evk_android_uuu_defconfig
+++ b/configs/imx8mp_evk_android_uuu_defconfig
@@ -58,7 +58,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_defconfig b/configs/imx8mp_evk_defconfig
index 7112fdf5859..70b23904a76 100644
--- a/configs/imx8mp_evk_defconfig
+++ b/configs/imx8mp_evk_defconfig
@@ -56,7 +56,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
diff --git a/configs/imx8mp_evk_inline_ecc_defconfig b/configs/imx8mp_evk_inline_ecc_defconfig
index 1c9b4177a6e..d1afc232f09 100644
--- a/configs/imx8mp_evk_inline_ecc_defconfig
+++ b/configs/imx8mp_evk_inline_ecc_defconfig
@@ -53,7 +53,7 @@ CONFIG_SYS_PROMPT="u-boot=> "
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HUSH_PARSER=y
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_evk_ndm_defconfig b/configs/imx8mp_evk_ndm_defconfig
index 83df7a1b94d..0f4d1207fbe 100644
--- a/configs/imx8mp_evk_ndm_defconfig
+++ b/configs/imx8mp_evk_ndm_defconfig
@@ -60,7 +60,7 @@ CONFIG_SPL_WATCHDOG=y
 CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_SYS_PROMPT="u-boot=> "
 # CONFIG_BOOTM_NETBSD is not set
 # CONFIG_CMD_EXPORTENV is not set
diff --git a/configs/imx8mp_rsb3720a1_4G_defconfig b/configs/imx8mp_rsb3720a1_4G_defconfig
index 14e6a3f3d05..b938a969213 100644
--- a/configs/imx8mp_rsb3720a1_4G_defconfig
+++ b/configs/imx8mp_rsb3720a1_4G_defconfig
@@ -31,7 +31,7 @@ CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_FIT_SIGNATURE=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx8mp_rsb3720a1_6G_defconfig b/configs/imx8mp_rsb3720a1_6G_defconfig
index fecdaed4979..a261afd5bcc 100644
--- a/configs/imx8mp_rsb3720a1_6G_defconfig
+++ b/configs/imx8mp_rsb3720a1_6G_defconfig
@@ -31,7 +31,7 @@ CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_FIT_SIGNATURE=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx8mq_cm_defconfig b/configs/imx8mq_cm_defconfig
index a5f8f0b9510..a2c310afaf4 100644
--- a/configs/imx8mq_cm_defconfig
+++ b/configs/imx8mq_cm_defconfig
@@ -25,7 +25,7 @@ CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_SPL_FIT_PRINT=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_PBSIZE=1050
diff --git a/configs/imx91_11x11_evk_defconfig b/configs/imx91_11x11_evk_defconfig
index c017ccc9ace..76982f1a708 100644
--- a/configs/imx91_11x11_evk_defconfig
+++ b/configs/imx91_11x11_evk_defconfig
@@ -29,6 +29,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx91_11x11_evk_inline_ecc_defconfig b/configs/imx91_11x11_evk_inline_ecc_defconfig
index d3e3200a23c..1a99715184c 100644
--- a/configs/imx91_11x11_evk_inline_ecc_defconfig
+++ b/configs/imx91_11x11_evk_inline_ecc_defconfig
@@ -30,6 +30,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx91_11x11_evk_spinand_defconfig b/configs/imx91_11x11_evk_spinand_defconfig
index 2ea6998c247..6e5b4899b7f 100644
--- a/configs/imx91_11x11_evk_spinand_defconfig
+++ b/configs/imx91_11x11_evk_spinand_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_android_defconfig b/configs/imx93_11x11_evk_android_defconfig
index 294740bb725..e92f4eb78ad 100644
--- a/configs/imx93_11x11_evk_android_defconfig
+++ b/configs/imx93_11x11_evk_android_defconfig
@@ -29,6 +29,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_android_dual_defconfig b/configs/imx93_11x11_evk_android_dual_defconfig
index d056161e43a..c0f2cbba0d7 100644
--- a/configs/imx93_11x11_evk_android_dual_defconfig
+++ b/configs/imx93_11x11_evk_android_dual_defconfig
@@ -29,6 +29,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_android_trusty_defconfig b/configs/imx93_11x11_evk_android_trusty_defconfig
index 97cbcb963de..bd8c5a771b7 100644
--- a/configs/imx93_11x11_evk_android_trusty_defconfig
+++ b/configs/imx93_11x11_evk_android_trusty_defconfig
@@ -29,6 +29,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_android_trusty_dual_defconfig b/configs/imx93_11x11_evk_android_trusty_dual_defconfig
index a45cb49c520..1701a4bc801 100644
--- a/configs/imx93_11x11_evk_android_trusty_dual_defconfig
+++ b/configs/imx93_11x11_evk_android_trusty_dual_defconfig
@@ -29,6 +29,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_android_uuu_defconfig b/configs/imx93_11x11_evk_android_uuu_defconfig
index e4db73029ca..d58d4e3915e 100644
--- a/configs/imx93_11x11_evk_android_uuu_defconfig
+++ b/configs/imx93_11x11_evk_android_uuu_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_defconfig b/configs/imx93_11x11_evk_defconfig
index 93b5c5a6575..287707466c8 100644
--- a/configs/imx93_11x11_evk_defconfig
+++ b/configs/imx93_11x11_evk_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_matter_trusty_defconfig b/configs/imx93_11x11_evk_matter_trusty_defconfig
index bf9357662ba..ab8b545033e 100644
--- a/configs/imx93_11x11_evk_matter_trusty_defconfig
+++ b/configs/imx93_11x11_evk_matter_trusty_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/imx93_11x11_evk_pf0900_defconfig b/configs/imx93_11x11_evk_pf0900_defconfig
index 95c90a66d16..b2ce1e0c0b9 100644
--- a/configs/imx93_11x11_evk_pf0900_defconfig
+++ b/configs/imx93_11x11_evk_pf0900_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
diff --git a/configs/kontron_pitx_imx8m_defconfig b/configs/kontron_pitx_imx8m_defconfig
index 02fc696b76f..b49f944a109 100644
--- a/configs/kontron_pitx_imx8m_defconfig
+++ b/configs/kontron_pitx_imx8m_defconfig
@@ -27,7 +27,7 @@ CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
 CONFIG_SPL_FIT_PRINT=y
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_SYS_CBSIZE=256
diff --git a/configs/msc_sm2s_imx8mp_defconfig b/configs/msc_sm2s_imx8mp_defconfig
index 88b97ad57d8..b4a2e1463fb 100644
--- a/configs/msc_sm2s_imx8mp_defconfig
+++ b/configs/msc_sm2s_imx8mp_defconfig
@@ -25,7 +25,7 @@ CONFIG_SYS_BARGSIZE=2048
 CONFIG_FIT=y
 CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DEFAULT_FDT_FILE="imx8mp-msc-sm2s-ep1.dtb"
diff --git a/configs/pico-imx8mq_defconfig b/configs/pico-imx8mq_defconfig
index b17a8d666e4..3732a814952 100644
--- a/configs/pico-imx8mq_defconfig
+++ b/configs/pico-imx8mq_defconfig
@@ -48,7 +48,7 @@ CONFIG_SPL_USB_GADGET=y
 CONFIG_SPL_USB_SDP_SUPPORT=y
 CONFIG_SYS_MAXARGS=64
 # CONFIG_BOOTM_NETBSD is not set
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
 # CONFIG_CMD_CRC32 is not set
diff --git a/configs/pico-imx93_defconfig b/configs/pico-imx93_defconfig
index 097b8c2737e..3be9e67bd0f 100644
--- a/configs/pico-imx93_defconfig
+++ b/configs/pico-imx93_defconfig
@@ -27,6 +27,8 @@ CONFIG_SYS_MEMTEST_START=0x80000000
 CONFIG_SYS_MEMTEST_END=0x90000000
 CONFIG_REMAKE_ELF=y
 CONFIG_FIT=y
+CONFIG_FIT_EXTERNAL_OFFSET=0x3000
+CONFIG_SPL_LOAD_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_DISTRO_DEFAULTS=y
diff --git a/configs/tek-imx8mp_defconfig b/configs/tek-imx8mp_defconfig
index d6c04488573..0d4a2e79b43 100644
--- a/configs/tek-imx8mp_defconfig
+++ b/configs/tek-imx8mp_defconfig
@@ -55,7 +55,7 @@ CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
 # CONFIG_BOOTM_NETBSD is not set
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
 CONFIG_CMD_ERASEENV=y
diff --git a/configs/tep-imx8mp_defconfig b/configs/tep-imx8mp_defconfig
index 36a55a382d7..275302f8b4d 100644
--- a/configs/tep-imx8mp_defconfig
+++ b/configs/tep-imx8mp_defconfig
@@ -55,7 +55,7 @@ CONFIG_SYS_MAXARGS=64
 CONFIG_SYS_CBSIZE=2048
 CONFIG_SYS_PBSIZE=2074
 # CONFIG_BOOTM_NETBSD is not set
-CONFIG_SYS_BOOTM_LEN=0x2000000
+CONFIG_SYS_BOOTM_LEN=0x4000000
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
 CONFIG_CMD_ERASEENV=y
-- 
2.25.1

